src/mds/MDCache.cc

```diff
@@ -12035,6 +12035,20 @@ void MDCache::show_subtrees(int dbl)
     }
   }
 
+  if (subtrees_seen.size() > SUBTREES_COUNT_THRESHOLD && depth >
+      SUBTREES_DEPTH_THRESHOLD) {
+    dout(15) << "number of subtrees > SUBTREES_COUNT_THRESHOLD and depth > "
+		"SUBTREES_DEPTH_THRESHOLD; not printing subtrees" << dendl;
+  } else if (subtrees_seen.size() > SUBTREES_COUNT_THRESHOLD) {
+    dout(15) << "number of subtrees > SUBTREES_COUNT_THRESHOLD; skipping "
+		"printing subtrees" << dendl;
+  } else if (depth > SUBTREES_DEPTH_THRESHOLD) {
+    dout(15) << "max depth among subtrees > SUBTREES_DEPTH_THRESHOLD; "
+		"not printing subtrees." << dendl;
+  }
+
+  if (!g_conf()->subsys.should_gather<ceph_subsys_mds, 25>())
+    return;
```

Comment:
Let's add a bool argument that forces the subtrees to be printed. We want the full subtree list probably every ~5 seconds via `MDBalancer::tick`.

Reply:
> Let's add a bool argument that forces the subtrees to be printed. 

I don't get this part.

Reply:
> > Let's add a bool argument that forces the subtrees to be printed.
> 
> I don't get this part.

Adding a bool variable that would print subtrees irrespective of their number/size. Right?

Reply:
> > > Let's add a bool argument that forces the subtrees to be printed.
> > 
> > 
> > I don't get this part.
> 
> Adding a bool variable that would print subtrees irrespective of their number/size. Right?

Yes, subject to the passed in `dbl`. i.e. instead do this:

`if (!force && !g_conf()->subsys.should_gather<ceph_subsys_mds, 25>())`